delphi-frontend/src/main/java/au/com/integradev/delphi/operator/OperatorInvocableCollector.java

```diff
@@ -439,6 +439,38 @@ private Set<Invocable> createIntegerArithmeticBinary(String name) {
     return result;
   }
 
+  private Set<Invocable> createSubtract() {
```

Comment:
This only differs from `createIntegerArithmeticBinary` in the result type for `uint31 <op> uint31`.
I'd say this should be reflected in the code (e.g. by pulling out the common code into a function that accepts the return type for the differing case), unless you have a particular reason for wanting the code to be flat.

Reply:
I can't think of an obvious way to abstract this logically, do you have any thoughts?

Reply:
My idea was basically just to have the return type of the `uint31 <op> uint31` intrinsic be a parameter to a the function which wraps the common code. Then you could have `createSubtract` just call that function and provide the output type for that intrinsic operator.

Reply:
Done.